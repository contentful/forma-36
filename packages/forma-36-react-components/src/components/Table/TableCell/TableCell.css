@import 'settings/colors';
@import 'settings/dimensions';
@import 'settings/typography';

.TableCell--head {
  background-color: var(--color-element-lightest);
  font-size: var(--font-size-s);
  font-weight: var(--font-weight-medium);
}

.TableCell--head__sorting {
  color: var(--color-text-dark);
}

.TableCell {
  border-bottom: 1px solid var(--color-element-light);
  color: var(--color-text-mid);
  font-size: var(--font-size-m);
  font-family: var(--font-stack-primary);
  font-weight: var(--font-weight-normal);
  line-height: 1.5;
  padding: var(--spacing-s) var(--spacing-m);
  vertical-align: top;
}

tr:hover td:not(.TableCell--head) {
  background-color: var(--color-element-lightest);
}

.TableRow--selected .TableCell {
  background-color: var(--color-blue-lightest);
}

.TableRow--selected .TableCell:last-child {
  box-shadow: inset -2px 0 0 var(--color-blue-mid);
}

tbody tr:last-child .TableCell {
  border-bottom: none;
}
