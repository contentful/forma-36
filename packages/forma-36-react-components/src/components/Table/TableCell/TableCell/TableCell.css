@import 'settings/colors';
@import 'settings/dimensions';
@import 'settings/typography';

.TableCell {
  border-bottom: 1px solid var(--color-element-light);
  color: var(--color-text-mid);
  font-size: var(--font-size-m);
  font-family: var(--font-stack-primary);
  font-weight: var(--font-weight-normal);
  line-height: 1.5;
  padding: var(--spacing-s) var(--spacing-m);
  vertical-align: top;
}

tbody tr:last-child .TableCell {
  border-bottom: none;
}

.TableCell--head {
  background-color: var(--color-element-lightest);
  font-size: var(--font-size-s);
  font-weight: var(--font-weight-medium);
}

.TableCell--head__sorting {
  color: var(--color-text-dark);
}
