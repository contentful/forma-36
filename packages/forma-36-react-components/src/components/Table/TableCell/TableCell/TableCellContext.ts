import React from 'react';

export const contextOptions = {
  body: { name: 'body', element: 'td', offsetTop: 0 },
  head: { name: 'head', element: 'th', offsetTop: 0 },
};

export const TableCellContext = React.createContext(contextOptions.body);
