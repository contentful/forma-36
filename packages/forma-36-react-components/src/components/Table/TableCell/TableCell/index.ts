export { default } from './TableCell';
export { TableCellContext, contextOptions } from './TableCellContext';
