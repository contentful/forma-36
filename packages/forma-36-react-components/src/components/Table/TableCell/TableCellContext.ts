import React from 'react';

export const contextOptions = {
  body: { name: 'body', element: 'td', offsetTop: 0 as number | string },
  head: { name: 'head', element: 'th', offsetTop: 0 as number | string },
};

export const TableCellContext = React.createContext(contextOptions.body);
