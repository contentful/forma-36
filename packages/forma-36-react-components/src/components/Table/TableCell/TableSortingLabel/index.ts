export * from './TableSortingLabel';
export { default } from './TableSortingLabel';
