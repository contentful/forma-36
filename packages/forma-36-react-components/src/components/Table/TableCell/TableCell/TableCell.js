import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';

import { TableCellContext } from './TableCellContext';

import styles from './TableCell.css';

export const sortingDirections = {
  asc: 'asc',
  desc: 'desc',
};

export class TableCell extends React.Component {
  static propTypes = {
    extraClassNames: PropTypes.string,
    children: PropTypes.node,
    sorting: PropTypes.oneOf([false, ...Object.keys(sortingDirections)]),
    align: PropTypes.oneOf(['center', 'left', 'right']),
  };

  static defaultProps = {
    children: null,
    extraClassNames: '',
    sorting: false,
    align: 'left',
  };

  render() {
    const {
      extraClassNames,
      children,
      sorting,
      align,
      ...otherProps
    } = this.props;

    return (
      <TableCellContext.Consumer>
        {({ name: context, element: Element, offsetTop }) => (
          <Element
            className={cn(styles.TableCell, extraClassNames, {
              [styles['TableCell--head']]: context === 'head',
              [styles['TableCell--head__sorting']]: sorting,
            })}
            style={{
              top: offsetTop,
            }}
            align={align}
            {...otherProps}
          >
            {children}
          </Element>
        )}
      </TableCellContext.Consumer>
    );
  }
}

export default TableCell;
