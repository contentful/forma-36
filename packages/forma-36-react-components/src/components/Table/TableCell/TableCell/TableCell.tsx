import React, { HTMLProps } from 'react';
import cn from 'classnames';

import { TableCellContext } from './TableCellContext';

import styles from './TableCell.css';
import { ElementType } from 'react';

export const sortingDirections = {
  asc: 'asc',
  desc: 'desc',
};

export type TableCellSorting = keyof typeof sortingDirections | boolean;

export interface TableCellProps extends HTMLProps<HTMLTableCellElement> {
  align?: 'center' | 'left' | 'right';
  sorting?: TableCellSorting;
  className?: string;
  testId?: string;
  children?: React.ReactNode;
}

const defaultProps: Partial<TableCellProps> = {
  align: 'left',
  sorting: false as TableCellSorting,
  testId: 'cf-ui-table-cell',
};

const TableCell = (props: TableCellProps) => {
  const { className, children, sorting, align, testId, ...otherProps } = props;

  return (
    <TableCellContext.Consumer>
      {({ name: context, element, offsetTop }) => {
        const Element = element as ElementType;
        return (
          <Element
            className={cn(styles['TableCell'], className, {
              [styles['TableCell--head']]: context === 'head',
              [styles['TableCell--head__sorting']]: sorting,
            })}
            style={{
              top: offsetTop,
            }}
            align={align}
            data-test-id={testId}
            {...otherProps}
          >
            {children}
          </Element>
        );
      }}
    </TableCellContext.Consumer>
  );
};

TableCell.defaultProps = defaultProps;

export default TableCell;
