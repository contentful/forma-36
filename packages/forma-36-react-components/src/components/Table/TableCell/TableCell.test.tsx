import React from 'react';
import { render } from '@testing-library/react';

import { TableCell } from './TableCell';

it('renders the component', () => {
  const table = document.createElement('table');
  const row = document.createElement('tr');
  document.body.appendChild(table);

  const { container } = render(<TableCell sorting="asc">Foo</TableCell>, {
    container: table.appendChild(row),
  });

  expect(container.firstChild).toMatchSnapshot();
});
