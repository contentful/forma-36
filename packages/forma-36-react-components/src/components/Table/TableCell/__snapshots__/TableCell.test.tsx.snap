// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
<td
  align="left"
  class="TableCell TableCell--head__sorting"
  data-test-id="cf-ui-table-cell"
  style="top: 0px;"
>
  Foo
</td>
`;
